File: Data Structures & Algorithms/anagram-groups/submission-6.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groupings = {}
        res = []
        for s in strs:
            key = ""
            counts = [0 for _ in range(26)]
            for c in s:
                counts[ord(c) - 97] += 1
            for count in counts:
                key += str(count) + "#"
            if key in groupings:
                groupings[key].append(s)
            else:
                groupings[key] = [s]
        
        for key,group in groupings.items():
            res.append(group)
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int i = 0, j = 0;
        while(j < prices.size()){
            if(prices[i] < prices[j]){
                res = max(res, prices[j] - prices[i]);
            }
            else{
                i = j;
            }
            j++;
        }
        return res;
    }
};
